jwst/resample/resample_spec.py

```diff
@@ -157,15 +157,20 @@ def build_interpolated_output_wcs(self, refmodel=None):
                 ra_center_pt = np.nanmean(ra_center)
                 dec_center_pt = np.nanmean(dec_center)
 
-                # ra and dec this converted to tangent projection
-                tan = Pix2Sky_TAN()
-                native2celestial = RotateNative2Celestial(ra_center_pt, dec_center_pt, 180)
-                undist2sky1 = tan | native2celestial
-                # Filter out RuntimeWarnings due to computed NaNs in the WCS
-                warnings.simplefilter("ignore")
-                # at this center of slit find x,y tangent proction - x_tan, y_tan
-                x_tan, y_tan = undist2sky1.inverse(ra, dec)
-                warnings.resetwarnings()
+                if model.meta.wcs.output_frame.name == 'world':
+                    # convert ra and dec to tangent projection
+                    tan = Pix2Sky_TAN()
+                    native2celestial = RotateNative2Celestial(ra_center_pt, dec_center_pt, 180)
+                    undist2sky1 = tan | native2celestial
+                    # Filter out RuntimeWarnings due to computed NaNs in the WCS
+                    warnings.simplefilter("ignore")
+                    # at this center of slit find x,y tangent projection - x_tan, y_tan
+                    x_tan, y_tan = undist2sky1.inverse(ra, dec)
+                    warnings.resetwarnings()
+                else:
```

Comment:
I wonder if this `else` statement is too general. If the general idea is to have coordinates on a plane and If the output_frame is `V2V3` the code may not work correctly since `V2V3` is a spherical system. I'm not sure if it's better to exclude `V2V3` or to consider only `msa_frame`. The first option allows for other frames to work so it may be prefferable.

Reply:
It's a tricky one - currently resample_spec only 'knows' how to do sky/world and msa_frame, so we could just test those cases and punt if there's a different output frame.  Alternatively, maybe we could do something like
`if Unit("deg") in model.meta.wcs.output_frame.unit or Unit("arcsec") in model.meta.wcs.output_frame.unit:`
to capture the sky-like cases, and otherwise do the cartesian cases